skrifa/src/variation.rs

```diff
@@ -0,0 +1,539 @@
+    }
+
+    /// Returns the maximum value of the axis.
+    pub fn max_value(&self) -> f32 {
+        self.record.max_value().to_f64() as _
+    }
+
+    /// Returns a normalized coordinate for the given user coordinate. The value will be
+    /// clamped to the range specified by the minimum and maximum values.
+    ///
+    /// This does not apply any axis variation remapping.
+    pub fn normalize(&self, coord: f32) -> NormalizedCoord {
+        self.record
+            .normalize(Fixed::from_f64(coord as _))
+            .to_f2dot14()
+    }
+}
+
+/// Collection of axes in a variable font.
+///
+/// See the [`Axis`] type for more detail.
+#[derive(Clone)]
+pub struct AxisCollection<'a> {
+    fvar: Option<Fvar<'a>>,
+    avar: Option<Avar<'a>>,
+}
+
+impl<'a> AxisCollection<'a> {
+    /// Creates a new axis collection from the given font.
+    pub fn new(font: &impl TableProvider<'a>) -> Self {
+        let fvar = font.fvar().ok();
+        let avar = font.avar().ok();
+        Self { fvar, avar }
+    }
+
+    /// Returns the number of variation axes in the font.
+    pub fn len(&self) -> usize {
+        self.fvar
+            .as_ref()
+            .map(|fvar| fvar.axis_count() as usize)
+            .unwrap_or(0)
+    }
+
+    /// Returns true if the collection is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns the axis at the given index.
+    pub fn get(&self, index: usize) -> Option<Axis> {
+        let raw = self.fvar.as_ref()?.axes().ok()?.get(index)?.clone();
+        Some(Axis { index, record: raw })
+    }
+
+    /// Returns the axis with the given tag.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use skrifa::prelude::*;
+    /// # fn wrapper(font: &FontRef) {
+    /// let opsz = Tag::new(b"opsz");
+    /// assert_eq!(font.axes().get_by_tag(opsz).unwrap().tag(), opsz);
+    /// # }
+    /// ```
+    pub fn get_by_tag(&self, tag: Tag) -> Option<Axis> {
+        self.iter().find(|axis| axis.tag() == tag)
+    }
+
+    /// Given an iterator of variation settings in user space, computes an
+    /// ordered sequence of normalized coordinates.
+    ///
+    /// * Setting selectors that don't match an axis are ignored.
+    /// * Setting values are clamped to the range of their associated axis
+    ///     before normalization.
+    /// * If more than one setting for an axis is provided, the last one is
+    ///     used.
+    /// * Omitted settings are set to 0.0, representing the default position
+    ///     in variation space.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use skrifa::prelude::*;
+    /// # fn wrapper(font: &FontRef) {
+    /// let location = font.axes().location(&[("wght", 250.0), ("wdth", 75.0)]);
+    /// # }
+    /// ```
+    pub fn location<I>(&self, settings: I) -> Location
+    where
+        I: IntoIterator,
+        I::IntoIter: 'a + Clone,
+        I::Item: Into<VariationSetting>,
```

Comment:
Yes, this is the same signature as the one in `ScalerBuilder` (which I'll probably just remove now that we have this method). I'm holding firm on this style because I think there is real value in being able to express variations in a low friction way. 

If you look at the tests (and doc comments) in this PR, you'll see the settings specified as something like
```rust
[("wght", 100.0), ("wdth", 75.0)]
```
 which I believe is simple, clear, and easy to understand for anyone familiar with the CSS `font-variation-settings` property. 

The alternative is
```rust
[VariationSetting::new(Tag::new(b"wght"), 100.0), VariationSetting::new(Tag::new(b"wdth"), 75.0)]
```
which is incredibly noisy and difficult to read. It is still useful to have the `VariationSetting` type though both as a place to attach documentation, and as a target for `Into`.

I did however notice that the `I::IntoIter: 'a + Clone` bound is unnecessary here so I'll remove that and it should make the signature more approachable. At least, I think `I: IntoIterator, I::Item: Into<T>` is a common enough pattern in Rust that it should be clear.

Reply:
yea the other alternative that jumps out at me is ignoring the iterator bit and just expecting it to be a slice? but just thinking out loud, if this works then great. :)